src/main/java/nextstep/sessions/domain/Session.java

```diff
@@ -0,0 +1,82 @@
+package nextstep.sessions.domain;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import nextstep.users.domain.NsUser;
+
+public class Session {
+
+    private Long id;
+
+    private int maxNumber;
+
+    private Status status;
+
+    private long price;
+
+    private LocalDateTime startedAt;
+
+    private LocalDateTime endAt;
+
+    private List<NsUser> attendees = new ArrayList<>();
+
+    public Session(final Status status) {
+        this(Integer.MAX_VALUE, status, 0L, LocalDateTime.now(), LocalDateTime.MAX);
+    }
+
+    public Session(final int maxNumber, final Status status, final long price, final LocalDateTime endAt) {
+        this(maxNumber, status, price, LocalDateTime.now(), endAt);
+    }
+
+    public Session(final int maxNumber, final Status status, final long price, final LocalDateTime startedAt,
+                   final LocalDateTime endAt) {
+        this.maxNumber = maxNumber;
+        this.status = status;
+        this.price = price;
+        this.startedAt = startedAt;
+        this.endAt = endAt;
+    }
+
+    public void enroll(final NsUser user, final long price) {
+        validate(price);
+        attendees.add(user);
+    }
+
+    private void validate(final long price) {
+        validatePrice(price);
+        validateAttendeeNumber();
+        validateRecruitingStatus();
+        validateStartedAt();
+    }
+
+    private void validatePrice(final long price) {
+        if (this.price != price) {
+            throw new IllegalArgumentException("가격이 일치하지 않습니다.");
+        }
+    }
+
+    private void validateAttendeeNumber() {
+        if (attendees.size() >= maxNumber) {
+            throw new IllegalArgumentException("수강 인원이 초과되었습니다.");
+        }
+    }
+
+    private void validateRecruitingStatus() {
+        if (status != Status.RECRUITING) {
```

Comment:
메시지를 보내서 처리해보면 어떨까요?
```suggestion
        if (status.isRecruiting()) {
```

Reply:
이게 더 객체 지향적인 설계겠네요!